src/app/public/modules/dropdown/dropdown.component.ts

```diff
@@ -204,91 +291,170 @@ export class SkyDropdownComponent implements OnInit, OnDestroy {
   }
 
   public ngOnDestroy(): void {
+    this.destroyAffixer();
+    this.destroyOverlay();
+
     this.ngUnsubscribe.next();
     this.ngUnsubscribe.complete();
+    this.ngUnsubscribe = undefined;
   }
 
-  @HostListener('keydown', ['$event'])
-  public onKeyDown(event: KeyboardEvent): void {
-    const key = event.key.toLowerCase();
+  private addEventListeners(): void {
+    const buttonElement = this.triggerButton.nativeElement;
 
-    if (this._isOpen) {
-      /* tslint:disable-next-line:switch-default */
-      switch (key) {
-        // After an item is selected with the enter key,
-        // wait a moment before returning focus to the dropdown trigger element.
-        case 'enter':
-          this.windowRef.getWindow().setTimeout(() => {
-            this.sendMessage(SkyDropdownMessageType.FocusTriggerButton);
-          });
-          break;
+    Observable
+      .fromEvent(buttonElement, 'click')
+      .takeUntil(this.ngUnsubscribe)
+      .subscribe(() => {
+        if (this.isOpen) {
+          this.sendMessage(SkyDropdownMessageType.Close);
+        } else {
+          this.sendMessage(SkyDropdownMessageType.Open);
+        }
+      });
 
-        // Allow the menu to be opened with the arrowdown key
-        // if it is first opened with the mouse.
-        case 'down':
-        case 'arrowdown':
-          if (!this.isKeyboardActive) {
-            this.isKeyboardActive = true;
+    Observable
+      .fromEvent(buttonElement, 'keyup')
+      .takeUntil(this.ngUnsubscribe)
+      .subscribe((event: KeyboardEvent) => {
+        const key = event.key.toLowerCase();
+        if (this.isOpen && key === 'escape') {
+          this.sendMessage(SkyDropdownMessageType.Close);
+          this.sendMessage(SkyDropdownMessageType.FocusTriggerButton);
+          event.stopPropagation();
```

Comment:
If you put a dropdown menu in a popover and the user presses the escape key, it would close both the dropdown and the popover in one keypress (unless we stop the propagation). I couldn't figure out a way to do it without stopping propagation unless you have an idea?

Reply:
Yeah, looking at the rest of my review, I think I also realized this challenge when looking at your `SkyPopoverDirective` changes. It's necessary there too. Since this doesn't take into account any click targets or focus, I feel like the only way to do this is with the `isOpen` check and then stop propagation. Let's keep it.